tensorflow/c/experimental/filesystem/modular_filesystem_test.cc

```diff
@@ -85,17 +85,34 @@ class ModularFileSystemTest : public ::testing::TestWithParam<std::string> {
     const std::string test_name = tensorflow::str_util::StringReplace(
         ::testing::UnitTest::GetInstance()->current_test_info()->name(), "/",
         "_", /*replace_all=*/true);
+    // Since we need the tests for cloud filesystem to run on all OSs (Windows,
+    // MacOS, Linux, ...) The path to temp directory must not be dependent on
+    // the OS which runs the tests.
+    const std::string tmp_dir_ =
+        cloud_path_.empty() ? ::testing::TempDir() : "/tmp/";
```

Comment:
`::testing::TempDir()` includes `/tmp` on its output. https://github.com/google/googletest/blob/0eea2e9fc63461761dea5f2f517bd6af2ca024fa/googletest/src/gtest.cc#L6401-L6425

We should instead make this a flag which if not set will default to `::testing::TempDir()`.

Reply:
if this test run on Windows, `::testing::TempDir()` will return `\\tmp\\`, so this test won't work